sale_order_line_discount_switch/models/sale_order_line.py

```
@@ -0,0 +1,94 @@
+            vals["discount_fixed"] = 0.0
+        will_convert_percent_to_fixed = bool(
+            vals.get("discount_mode") == "fixed" and "discount_fixed" not in vals
+        )
+        lines_to_fill_fixed = self.env["sale.order.line"]
+        if will_convert_percent_to_fixed:
+            lines_to_fill_fixed = self.filtered(
+                lambda line: line.discount_mode == "percent"
+            )
+        res = super().write(vals)
+        for line in lines_to_fill_fixed:
+            if not line.discount_mode == "fixed":
```

Ruff (0.15.10)

[warning] Use `line.discount_mode != "fixed"` instead of `not line.discount_mode == "fixed"`

Replace with `!=` operator

(SIM201)